python/cudf/cudf/core/series.py

```diff
@@ -1437,10 +1437,23 @@ def astype(self, dtype, **kwargs):
             Copy of ``self`` cast to the given dtype. Returns
             ``self`` if ``dtype`` is the same as ``self.dtype``.
         """
+        if errors not in ("ignore", "raise"):
+            raise ValueError("invalid error value specified")
+
         if pd.api.types.is_dtype_equal(dtype, self.dtype):
             return self
-
-        return self._copy_construct(data=self._column.astype(dtype, **kwargs))
+        try:
+            return self._copy_construct(
+                data=self._column.astype(dtype, **kwargs)
+            )
+        except Exception as e:
+            if errors == "raise":
+                raise e
+            elif errors == "ignore":
+                warnings.warn(
```

Comment:
Why does the message differ depending on raise/ignore? Should we use `warnings.warn(e.message)`, or maybe create a new error to raise?

Reply:
Good question. My thinking was that since the casting is done at the series level, it makes sense to propagate the series level error back up to the user if an error occurs in the `raise` case. In the `ignore` case, my thoughts were just to say something hopefully informative to the user to warn them that some particular column couldn't be cast. We could emit a warning that contains the full error message if that seems more helpful.

Reply:
I think it would be better for the warning/error to be identical, since the only difference is the how severe the consumer considers the exception circumstance, and not the actual circumstance itself.

Reply:
makes sense to me: `raise` -> raise the error, `ignore` -> log the error and keep going.

Reply:
Now that you put it that way (and I don't feel strongly about this, either way), since it's a breaking change, I'll throw in `warn` as an option, and `ignore` literally just does nothing. But to be clear, consistency > accuracy, imho, so if we're already using `ignore` for warnings in other places, we should go with that.

Reply:
I don't see anywhere else in the python code at least where we are using `ignore`, and I like your idea, so I'll add that as an option :)